auter

```diff
@@ -301,6 +323,21 @@ if [[ -z "${CONFIGSET}" ]]; then
   quit 5
 fi
 
+if [[ "${ONLYINSTALLFROMPREP}" == "yes" ]]; then
+  if [[ ! -d "${DOWNLOADDIR}/${CONFIGSET}" ]]; then
+    install -m 755 -o root -g root -d ${DOWNLOADDIR}/${CONFIGSET}
+  elif [[ $(stat -c %G%U%a "${DOWNLOADDIR}") != rootroot[0-9][0-9][0,1,4,5] ]]; then
+    logit "ERROR: ${DOWNLOADDIR}/${CONFIGSET} does not have the correct permissions."
+    quit 3
+  fi
+  DOWNLOADOPTION="--downloaddir=${DOWNLOADDIR}/${CONFIGSET}"
```

Comment:
I don't like setting DOWNLOADOPTION here, when we're actually only using the variable within the prep_updates function.  Could we move this to the logic where it's actually being used?

Even better would be to just use the download to cache and move to the final location method for both yum and dnf - reduces code complexity, even if it's not quite as elegant as --downloaddir.

Same story for setting the UPDATEACTION variable.  Logic related to what we do to prep/apply updates should generally be in the relevant function instead of main().